coderd/database/dbmem/dbmem.go

```diff
@@ -4853,6 +4863,22 @@ func (q *FakeQuerier) GetUserLinksByUserID(_ context.Context, userID uuid.UUID)
 	return uls, nil
 }
 
+func (q *FakeQuerier) GetUserNotificationPreferences(_ context.Context, userID uuid.UUID) ([]database.NotificationPreference, error) {
+	q.mutex.RLock()
+	defer q.mutex.RUnlock()
+
+	out := make([]database.NotificationPreference, 0, len(q.notificationPreferences))
+	for _, np := range q.notificationPreferences {
+		if np.UserID != userID {
+			continue
+		}
+
+		out = append(out, np)
+	}
+
+	return out, nil
```

Comment:
`sql.ErrNoRows`?

Reply:
In this case, an empty result is what the database is returning.
I think `sql.ErrNoRows` is returned for `:one` or `:exec` etc, not for `:many` right?

Reply:
Ah fair 👍